src/cli/cli.ts

```diff
@@ -0,0 +1,110 @@
+import { mkdirSync, readdirSync } from 'fs';
+import { join } from 'path';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import prompts, { PromptObject } from 'prompts';
+import { copySync } from 'fs-extra';
+import chalk from 'chalk';
+import { go, isGoSuccess } from '../utils';
+
+const promptQuestions = (questions: PromptObject[]) =>
+  prompts(questions, {
+    // https://github.com/terkelg/prompts/issues/27#issuecomment-527693302
+    onCancel: () => {
+      throw new Error('Aborted by the user');
+    },
+  });
+
+const createCliOption = (name: string) => ({
+  title: name,
+  value: name,
+});
+
+const TEMPLATE_DIR = join(__dirname, '../../templates');
+export const getAvailableTemplates = () => readdirSync(TEMPLATE_DIR);
+export const getTemplatePath = (template: string) => join(TEMPLATE_DIR, template);
+
+export const createProjectUsingTemplate = (path: string, template: string) => {
+  console.info(`Creating a new beacon reader app in ${chalk.green(path)}`);
+
+  mkdirSync(path, { recursive: true });
+  copySync(getTemplatePath(template), path);
+
+  console.info(`Successfully created and initialized a new project.`);
+};
+
+export const createCli = () =>
+  yargs(hideBin(process.argv))
+    .command(
+      // The name of the cli binary is "create-beacon-reader-app" (see package.json) and it should generate the app
+      // without the need for additional command. This is inspired by how "create-react-app" works.
+      //
+      // See: https://github.com/yargs/yargs/blob/main/docs/advanced.md#default-commands
+      '$0',
+      'Create a project skeleton for an application that reads a value from beacon',
+      {
+        path: {
+          description:
+            "Path to a directory in which to create a project. The directory will be created if it doesn't exist yet.",
+          demandOption: false,
+          type: 'string',
+        },
+        template: {
+          description:
+            'The name of the template to be used. Use the "show-available-templates" command to list the available options.',
+          demandOption: false,
+          type: 'string',
+        },
```

Comment:
Please, add short options such as `-p` for `--path` and `-t` for `--template`.

Reply:
I actually dislike that - I always have to wonder what those shorthand's stand for.

Btw. `create-reacta-app` doesn't support that either.

Reply:
I know you dislike that, you don't have to use it :smile: But it is a common practice, it costs nothing, and users that use that (like me) can.

Reply:
Ok, I'll do it for you :D